File: DiscountsAPI/Models/Discount.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DiscountsAPI.Models
{
    public class Discount
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string DiscountId { get; set; }
        public string CouponCode { get; set; }
        public DateTime ExpiryDate { get; set; }
        public bool IsActive { get; set; }
        public int DiscountPercentage { get; set; }
    }
}

File: jeanstationfrontend/src/app/models/Discount.ts
export default class Discount{
    discountId: string;
    couponCode: string;
    expiryDate: Date;
    isActive: any;
    discountPercentage: number;

    constructor(){
        this.discountId = "";
        this.couponCode = "";
        this.expiryDate = new Date();
        this.isActive = true; 
        this.discountPercentage = 0;
    }
}